### PR TITLE
Enable IDE0055 (formatting analyzer) with CodeStyle NuGet package in VS IDE and CI

### PR DESCRIPTION
**Pull request by mavasani**
Formatting violations will now be reported as Suggestions (with ...) while typing in the editor. These violations will be escalated to warnings during local build and correspondingly **break build in CI** (executed with /warnaserror), similar to other analyzer diagnostics enabled on the repo.

### CODE REVIEW

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public abstract class PropagateCancellationTokensWhenPossibleAnalyzer : Diagnost
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs]: nit: remove comment

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/UsePreferredTerms.cs [hunk 2]
```diff
@@ -180,7 +180,7 @@ public abstract class UsePreferredTermsAnalyzer : DiagnosticAnalyzer
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(AssemblyRule, NamespaceRule, MemberParameterRule, DelegateParameterRule, TypeTypeParameterRule, MethodTypeParameterRule, TypeRule, MemberRule, AssemblyNoAlternateRule, NamespaceNoAlternateRule, MemberParameterNoAlternateRule, DelegateParameterNoAlternateRule, TypeTypeParameterNoAlternateRule, MethodTypeParameterNoAlternateRule, TypeNoAlternateRule, MemberNoAlternateRule);
+        //ImmutableArray.Create(AssemblyRule, NamespaceRule, MemberParameterRule, DelegateParameterRule, TypeTypeParameterRule, MethodTypeParameterRule, TypeRule, MemberRule, AssemblyNoAlternateRule, NamespaceNoAlternateRule, MemberParameterNoAlternateRule, DelegateParameterNoAlternateRule, TypeTypeParameterNoAlternateRule, MethodTypeParameterNoAlternateRule, TypeNoAlternateRule, MemberNoAlternateRule);
```
Comment (genlu) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/UsePreferredTerms.cs]: nit: remove comment

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiReview/AvoidCallingProblematicMethods.cs [hunk 3]
```diff
@@ -121,7 +121,7 @@ public abstract class AvoidCallingProblematicMethodsAnalyzer : DiagnosticAnalyze
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            // ImmutableArray.Create(SystemGCCollectRule, SystemThreadingThreadResumeRule, SystemThreadingThreadSuspendRule, SystemTypeInvokeMemberRule, CoInitializeSecurityRule, CoSetProxyBlanketRule, SystemRuntimeInteropServicesSafeHandleDangerousGetHandleRule, SystemReflectionAssemblyLoadFromRule, SystemReflectionAssemblyLoadFileRule, SystemReflectionAssemblyLoadWithPartialNameRule);
+        // ImmutableArray.Create(SystemGCCollectRule, SystemThreadingThreadResumeRule, SystemThreadingThreadSuspendRule, SystemTypeInvokeMemberRule, CoInitializeSecurityRule, CoSetProxyBlanketRule, SystemRuntimeInteropServicesSafeHandleDangerousGetHandleRule, SystemReflectionAssemblyLoadFromRule, SystemReflectionAssemblyLoadFileRule, SystemReflectionAssemblyLoadWithPartialNameRule);
```
Comment (genlu) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiReview/AvoidCallingProblematicMethods.cs]: nit: remove comment

File: src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/VariableNamesShouldNotMatchFieldNames.cs [hunk 4]
```diff
@@ -40,7 +40,7 @@ public abstract class VariableNamesShouldNotMatchFieldNamesAnalyzer : Diagnostic
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(LocalRule, ParameterRule);
+        //ImmutableArray.Create(LocalRule, ParameterRule);
```
Comment (genlu) [inline src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/VariableNamesShouldNotMatchFieldNames.cs]: nit: remove comment

File: src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/ReviewVisibleEventHandlers.cs [hunk 5]
```diff
@@ -40,7 +40,7 @@ public abstract class ReviewVisibleEventHandlersAnalyzer : DiagnosticAnalyzer
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(SecurityRule, DefaultRule);
+        //ImmutableArray.Create(SecurityRule, DefaultRule);
```
Comment (genlu) [inline src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/ReviewVisibleEventHandlers.cs]: nit: remove comment

File: src/Microsoft.NetCore.Analyzers/Core/InteropServices/MarkBooleanPInvokeArgumentsWithMarshalAs.cs [hunk 6]
```diff
@@ -40,7 +40,7 @@ public abstract class MarkBooleanPInvokeArgumentsWithMarshalAsAnalyzer : Diagnos
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(DefaultRule, ReturnRule);
+        //ImmutableArray.Create(DefaultRule, ReturnRule);
```
Comment (genlu) [inline src/Microsoft.NetCore.Analyzers/Core/InteropServices/MarkBooleanPInvokeArgumentsWithMarshalAs.cs]: nit: remove comment

File: src/Microsoft.NetCore.Analyzers/Core/InteropServices/UseManagedEquivalentsOfWin32Api.cs [hunk 7]
```diff
@@ -30,7 +30,7 @@ public abstract class UseManagedEquivalentsOfWin32ApiAnalyzer : DiagnosticAnalyz
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetCore.Analyzers/Core/InteropServices/UseManagedEquivalentsOfWin32Api.cs]: nit: remove comment

File: src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeMethodsShouldCallBaseClassDispose.cs [hunk 8]
```diff
@@ -30,7 +30,7 @@ public abstract class DisposeMethodsShouldCallBaseClassDisposeAnalyzer : Diagnos
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeMethodsShouldCallBaseClassDispose.cs]: nit: remove comment

File: src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseTimersThatPreventPowerStateChanges.cs [hunk 9]
```diff
@@ -30,7 +30,7 @@ public abstract class DoNotUseTimersThatPreventPowerStateChangesAnalyzer : Diagn
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseTimersThatPreventPowerStateChanges.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/AvoidDuplicateAccelerators.cs [hunk 10]
```diff
@@ -30,7 +30,7 @@ public abstract class AvoidDuplicateAcceleratorsAnalyzer : DiagnosticAnalyzer
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/AvoidDuplicateAccelerators.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/CallBaseClassMethodsOnISerializableTypes.cs [hunk 11]
```diff
@@ -30,7 +30,7 @@ public abstract class CallBaseClassMethodsOnISerializableTypesAnalyzer : Diagnos
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/CallBaseClassMethodsOnISerializableTypes.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/DoNotMarkServicedComponentsWithWebMethod.cs [hunk 12]
```diff
@@ -30,7 +30,7 @@ public abstract class DoNotMarkServicedComponentsWithWebMethodAnalyzer : Diagnos
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/DoNotMarkServicedComponentsWithWebMethod.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/ImplementISerializableCorrectly.cs [hunk 13]
```diff
@@ -50,7 +50,7 @@ public abstract class ImplementISerializableCorrectlyAnalyzer : DiagnosticAnalyz
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX ? ImmutableArray.Create(DefaultRule, MakeVisibleRule, MakeOverridableRule) : ImmutableArray<DiagnosticDescriptor>.Empty;
+        //DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX ? ImmutableArray.Create(DefaultRule, MakeVisibleRule, MakeOverridableRule) : ImmutableArray<DiagnosticDescriptor>.Empty;
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/ImplementISerializableCorrectly.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/ImplementSerializationMethodsCorrectly.cs [hunk 14]
```diff
@@ -70,7 +70,7 @@ public abstract class ImplementSerializationMethodsCorrectlyAnalyzer : Diagnosti
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX ? ImmutableArray.Create(VisibilityRule, ReturnTypeRule, ParametersRule, GenericRule, StaticRule) : ImmutableArray<DiagnosticDescriptor>.Empty;
+        //DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX ? ImmutableArray.Create(VisibilityRule, ReturnTypeRule, ParametersRule, GenericRule, StaticRule) : ImmutableArray<DiagnosticDescriptor>.Empty;
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/ImplementSerializationMethodsCorrectly.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/MarkWindowsFormsEntryPointsWithStaThread.cs [hunk 15]
```diff
@@ -30,7 +30,7 @@ public abstract class MarkWindowsFormsEntryPointsWithStaThreadAnalyzer : Diagnos
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/MarkWindowsFormsEntryPointsWithStaThread.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/ProvideDeserializationMethodsForOptionalFields.cs [hunk 16]
```diff
@@ -39,8 +39,8 @@ public abstract class ProvideDeserializationMethodsForOptionalFieldsAnalyzer : D
                                                                              helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca2239-provide-deserialization-methods-for-optional-fields",
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>  ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(OnDeserializedRule, OnDeserializingRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
+        //ImmutableArray.Create(OnDeserializedRule, OnDeserializingRule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/ProvideDeserializationMethodsForOptionalFields.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/SetLocaleForDataTypes.cs [hunk 17]
```diff
@@ -30,7 +30,7 @@ public abstract class SetLocaleForDataTypesAnalyzer : DiagnosticAnalyzer
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/SetLocaleForDataTypes.cs]: nit: remove comment

File: src/Microsoft.NetFramework.Analyzers/Core/SpecifyMessageBoxOptions.cs [hunk 18]
```diff
@@ -30,7 +30,7 @@ public abstract class SpecifyMessageBoxOptionsAnalyzer : DiagnosticAnalyzer
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(Rule);
+        //ImmutableArray.Create(Rule);
```
Comment (genlu) [inline src/Microsoft.NetFramework.Analyzers/Core/SpecifyMessageBoxOptions.cs]: nit: remove comment

File: src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs [hunk 19]
```diff
@@ -180,7 +180,7 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
...
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-            //ImmutableArray.Create(AssemblyRule, NamespaceRule, TypeRule, MemberRule, MemberParameterRule, DelegateParameterRule, TypeTypeParameterRule, MethodTypeParameterRule, AssemblyMoreMeaningfulNameRule, NamespaceMoreMeaningfulNameRule, TypeMoreMeaningfulNameRule, MemberMoreMeaningfulNameRule, MemberParameterMoreMeaningfulNameRule, DelegateParameterMoreMeaningfulNameRule, TypeTypeParameterMoreMeaningfulNameRule, MethodTypeParameterMoreMeaningfulNameRule);
+        //ImmutableArray.Create(AssemblyRule, NamespaceRule, TypeRule, MemberRule, MemberParameterRule, DelegateParameterRule, TypeTypeParameterRule, MethodTypeParameterRule, AssemblyMoreMeaningfulNameRule, NamespaceMoreMeaningfulNameRule, TypeMoreMeaningfulNameRule, MemberMoreMeaningfulNameRule, MemberParameterMoreMeaningfulNameRule, DelegateParameterMoreMeaningfulNameRule, TypeTypeParameterMoreMeaningfulNameRule, MethodTypeParameterMoreMeaningfulNameRule);
```
Comment (genlu) [inline src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs]: nit: remove comment

File: eng/Analyzers_NonShippingRules_Build.ruleset [hunk 20]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="Common diagnostic rules for all Roslyn Analyzer projects"
+         Description="This file contains diagnostic settings used by all Roslyn Analyzer projects. Projects that need specific settings should have their own rule set files that Include this one, and then make the necessary adjustments." 
+         ToolsVersion="14.0">
+  <Include Path=".\Analyzers_NonShippingRules.ruleset" Action="Default" />
```
[the remark this answers is not in the document]
Comment (mavasani) [reply eng/Analyzers_NonShippingRules_Build.ruleset]: `Analyzers_NonShippingRules.ruleset` already includes this ruleset: https://github.com/dotnet/roslyn-analyzers/blob/master/eng/Analyzers_NonShippingRules.ruleset#L4

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs]: In the past we have had instances where developers have forgotten to update `SupportedDiagnostics` when implementing a new analyzer, and unnecessarily spent multiple investigation cycles. I'll retain the comments as it doesn't do much harm.

**Comment by mavasani** [comment]
> By the way, do you know how to make codeflow shows space-only changes?

No idea - wonder if someone from @dotnet/roslyn-analysis knows about it.

**Comment by mavasani** [comment]
@sharwell https://github.com/dotnet/roslyn-analyzers/pull/2093/commits/7eb1f66421be14fa0dc52623f93b65111e09cc77 sets `TargetFrameworkMonikerAssemblyAttributesFileClean` to true, but that doesn't seem to have prevented formatting violations reported on VB AssemblyAttributes files. Any thoughts on how I can suppress these?

**Pull request merged**